integration/airflow/openlineage/airflow/utils.py

```diff
@@ -85,6 +87,20 @@ def make_key(job_name, run_id):
         return "openlineage_id_mapping-{}-{}".format(job_name, run_id)
 
 
+def to_json_encodable(task: "BaseOperator") -> Dict[str, object]:
+    def _task_encoder(obj):
+        if isinstance(obj, datetime.datetime):
+            return obj.isoformat()
+        elif isinstance(obj, AIRFLOW_DAG):
+            return {'dag_id': obj.dag_id,
+                    'tags': obj.tags,
+                    'schedule_interval': obj.schedule_interval}
+        else:
+            return str(obj)
+
+    return json.loads(json.dumps(task.__dict__, default=_task_encoder))
```

Comment:
we should probably have some failsafe in case this is too big.
otherwise this looks good to me. 
I'm assuming using this json.dumps function ensures we won't get in an infinite loop if there's a cycle.